rdflib/collection.py

```diff
@@ -208,6 +212,8 @@ def __iter__(self) -> Iterator[Node]:
         return self.graph.items(self.uri)
 
     def _end(self) -> Node:
+        if self.uri == RDF.nil:
```

Comment:
Can we move this check from _end down to after line 252 in `__iadd__`, I think it makes more sense to catch it there.
That way we can still use `_end()` when `nil` is deliberately added to the end of a list.